### DISCUSSION TITLE
Keep extras for all object types

### DISCUSSION THREAD

**Discussion by wayne-wu** [zeux/meshoptimizer#670] [General]
I see that currently the `-ke` flag only supports extras on nodes, materials, and meshes(?). I'm curious why we don't just support extras on all object types? Is it a performance reason, or complexity in implementation?

Thanks!

**Comment by zeux** [comment]
The considerations are a little separate for different types. In general, gltfpack performs a full scene reassembly. Handling extras for some types is simple if they don't have special processing, but for some it requires consideration around whether it's safe to merge two different objects or if extras need to be taken into account when removing unused objects. Additionally glTF in theory supports extras everywhere, including on internal JSON objects that don't always correspond to an obvious scene level concept, eg extras can be present on the accessor or sparse object within the accessor.

So the general approach is to only add support when it's requested with a specific use case and when it doesn't interfere too much with scene processing.
